src/llmcompressor/transformers/compression/helpers.py

```diff
@@ -107,21 +107,26 @@ def hessian_memory_requirements(model: torch.nn.Module) -> int:
     :return: number of bytes required to reserve for GPTQ on a single layer
     """
     transformer_layers = get_layers(get_no_split_params(model), model)
-    single_layer = transformer_layers[list(transformer_layers.keys())[0]]
-    total_hessian_elems = 0
-    max_column_size = 0
-    for _, module in single_layer.named_modules():
-        if isinstance(module, Linear):
-            for param in module.parameters():
-                column_size = param.shape[1]
-                total_hessian_elems += column_size * column_size
-                if column_size > max_column_size:
-                    # max extra memory for inverse calculation
-                    max_column_size = column_size
-
+    total_hessian_elems = {}
```

Comment:
just so I understand, instead of looking at just a single layer, we're looking at a list of layers, given by the `get_layers` function?

Reply:
Yep, we need to find the biggest layer and largest column to get the total max